### PR TITLE
Frosty Bulk Exporter

### PR DESCRIPTION
**Pull request by IPdotSetAF**
This shall bring the Bulk Exporter plugin into the FrostyEditor.

- `GroupBox` template is added to the Editor theme.
- `FrostyDataExplorer` can have a `ContextMenu` on its `TreeView`.
- the selected path on the `DataExplorer` of `MainWindow` is statically stored in the `App` class.

### CODE REVIEW

File: README.md
Comment (HarGabt) [inline README.md:1]: Why are you replacing Frosty's readme with plugin's readme?

Comment (IPdotSetAF) [reply README.md:1]: maybe i better put the readme inside the plugin folder?

**Comment by shoushou1106** [comment]
Can u make a pr to 1.0.6.4 branch?

**Comment by IPdotSetAF** [comment]
> Can u make a pr to 1.0.6.4 branch?

Yeah sure

**Comment by wannkunstbeikor** [comment]
u should be able to get the SelectedPath through `App.EditorWindow.DataExplorer.SelectedPath`, so u wouldnt need to modify frosty core

**Comment by wannkunstbeikor** [comment]
and then i would recommend to create a separate repo for the plugin, since v1 is not receiving any major updates anymore and including a new untested plugin in a minor release is not sth i would like to do

**Comment by IPdotSetAF** [comment]
> u should be able to get the SelectedPath through `App.EditorWindow.DataExplorer.SelectedPath`, so u wouldnt need to modify frosty core

Thats fixable.


> and then i would recommend to create a separate repo for the plugin, since v1 is not receiving any major updates anymore and including a new untested plugin in a minor release is not sth i would like to do

I don't think a plugin would require a major version.
I understand that version 1 is not in development anymore but forcefully stopping any already developed feature of bugfix for version 1 is not logical (since its  currently the main and only available/usable version that people use)

**Comment by wannkunstbeikor** [comment]
> I understand that version 1 is not in development anymore but forcefully stopping any already developed feature of bugfix for version 1 is not logical

yeah but a plugin != bugfix

**Comment by wannkunstbeikor** [comment]
> I don't think a plugin would require a major version.

u can just release the plugin on ur own, which would make things easier for u and me, u can update it whenever u want, i dont have some untested plugin in a release and dont need to help people that have issues with it

**Comment by IPdotSetAF** [comment]
> > I understand that version 1 is not in development anymore but forcefully stopping any already developed feature of bugfix for version 1 is not logical
> 
> yeah but a plugin != bugfix

By bug fix i am referring to other open PRs. Of course i know this PR is not a bugfix.

**Comment by IPdotSetAF** [comment]
> > I don't think a plugin would require a major version.
> 
> u can just release the plugin on ur own, which would make things easier for u and me, u can update it whenever u want, i dont have some untested plugin in a release and dont need to help people that have issues with it

The point is that I can't, there are other essential core modifications, which i think you skipped. I can not just release a plugin that is only compatible with its own UNVERSIONED frosty editor/core

**Comment by IPdotSetAF** [comment]
As i said, I have also added a context menu to the tree view of the frostyDataExplorer

**Comment by wannkunstbeikor** [comment]
> The point is that I can't, there are other essential core modifications, which i think you skipped. I can not just release a plugin that is only compatible with its own UNVERSIONED frosty editor/core

u dont need the context menu, just create a menu extension for the bulk export and get the selected path like i told u

**Comment by wannkunstbeikor** [comment]
> By bug fix i am referring to other open PRs. Of course i know this PR is not a bugfix.

there are no open PRs which have bugfixes and are ready to be merged

**Comment by IPdotSetAF** [comment]
> u should be able to get the SelectedPath through `App.EditorWindow.DataExplorer.SelectedPath`, so u wouldnt need to modify frosty core

just reverted this one.

> > The point is that I can't, there are other essential core modifications, which i think you skipped. I can not just release a plugin that is only compatible with its own UNVERSIONED frosty editor/core
> 
> u dont need the context menu, just create a menu extension for the bulk export and get the selected path like i told u

there is the menu extension as well. the plugin Tab also uses the frostyDataEditor for selection which is also using the treeView context menu. 

the treeView Context Menu feature is added to the core in a non breaking way (made sure other plugins wont face any issue)

that being said, I'm sure this feature is not possible without core modifications as the other guy did in the BatchExporter thing.

And as the core is modified I cannot release the plugin on my own, because it wont be compatible with frosty unless the core changes are merged and versioned.

I hope this clears things up.

**Comment by wannkunstbeikor** [comment]
im 100% sure u can create a bulk exporter without modifying core frosty, since u only need the selected path and a button or sth which u can just use the menu extensions like i said, and then u can show a window for settings or whatever

**Comment by IPdotSetAF** [comment]
how am i supposed to know which path the user wants to export??
again, As I said, I HAVE already added the menu extension as well, under the `View` menu.
just read the readme inside the plugin folder and give it a test.

**Comment by wannkunstbeikor** [comment]
> how am i supposed to know which path the user wants to export??

use the selectedpath duh

**Comment by MagixGames** [comment]
wannkunst trying to be a nice and inviting project lead challenge

**Pull request closed without merging**